v2/dtos/event.go

```diff
@@ -18,11 +18,11 @@ type Event struct {
 	common.Versionable `json:",inline"`
 	ID                 string        `json:"id"`
 	Pushed             int64         `json:"pushed,omitempty"`
-	DeviceName         string        `json:"deviceName"`
+	DeviceName         string        `json:"deviceName" validate:"required"`
 	Created            int64         `json:"created"`
 	Modified           int64         `json:"modified,omitempty"`
 	Origin             int64         `json:"origin"`
-	Readings           []BaseReading `json:"readings"`
+	Readings           []BaseReading `json:"readings" validate:"required"`
```

Comment:
I think this needs to be: `validate:"required,gt=0,dive"`

Reply:
Updated.